dev-tools/xtask/src/a4x2_deploy.rs

```diff
@@ -0,0 +1,472 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Subcommand: cargo xtask a4x2-deploy
+
+use anyhow::{anyhow, bail, Result};
+use camino::Utf8PathBuf;
+use clap::{Args, Parser, Subcommand};
+use serde_json::Value;
+use std::env;
+use std::fs;
+use std::{thread, time};
+use xshell::{cmd, Shell};
+
+/// Args for sshing in without checking/storing the remote host key. Every time
+/// we start a4x2, it will have new keys.
+static INSECURE_SSH_ARGS: [&str; 8] = [
+    "-o",
+    "StrictHostKeyChecking no",
+    "-o",
+    "UserKnownHostsFile /dev/null",
+    "-o",
+    "GlobalKnownHostsFile /dev/null",
+    "-o",
+    "LogLevel error",
+];
+
+/// default path used when none provided on command line
+static DEFAULT_A4X2_PKG_PATH: &str = "out/a4x2-package-out.tgz";
+
+#[derive(Parser)]
+pub struct A4x2DeployArgs {
+    #[command(subcommand)]
+    command: DeployCommand,
+}
+
+#[derive(Subcommand, Clone)]
+pub enum DeployCommand {
+    /// Start a4x2, deploy a control plane to it, and then leave it running.
+    Start(StartArgs),
+
+    /// Stop a4x2 that was previously launched with the `start` subcommand.
+    Stop,
+
+    /// Start a4x2, run tests, and then stop a4x2 unless you request otherwise.
+    RunTests(RunTestsArgs),
+}
+
+#[derive(Args, Clone)]
+pub struct StartArgs {
+    /// Path to an a4x2-package bundle generated by a4x2-package xtask.
+    #[clap(long, default_value_t = Utf8PathBuf::from(DEFAULT_A4X2_PKG_PATH))]
+    package: Utf8PathBuf,
+}
+
+#[derive(Args, Clone)]
+pub struct RunTestsArgs {
+    /// Execute omicron live tests
+    #[clap(long)]
+    live_tests: bool,
+
+    /// Leave a4x2 running after tests. Useful for investigating test
+    /// failures.
+    #[clap(long)]
+    leave_running: bool,
+
+    /// Path to an a4x2-package bundle generated by a4x2-package xtask.
+    #[clap(long, default_value_t = Utf8PathBuf::from(DEFAULT_A4X2_PKG_PATH))]
+    package: Utf8PathBuf,
+}
+
+struct Environment {
+    a4x2_package_tar: Utf8PathBuf,
+    work_dir: Utf8PathBuf,
+    a4x2_dir: Utf8PathBuf,
+    out_dir: Utf8PathBuf,
+    in_ci: bool,
+}
+
+pub fn run_cmd(args: A4x2DeployArgs) -> Result<()> {
+    let sh = Shell::new()?;
+
+    // TODO: do we want to have a configurable falcon dataset?
+
+    let env = {
+        // Buildomat sets the CI environment variable. We'll work a bit
+        // differently in that case (no cleanup, use /work)
+        let in_ci = env::var("CI").is_ok();
```

Comment:
Flags like this make me worried about diverging behavior between CI and local operation.  How about allowing the work directory to be specified on the command line and having CI specify it?  I'm not that worried about skipping cleanup in CI but if we that we could have a `--skip-cleanup` arg, too.

Reply:
Having read through the rest of this PR, I'm coming around to believing that maybe configuration profiles wouldn't be the worst after all.